opentelemetry/src/global/trace.rs

```diff
@@ -51,6 +51,13 @@ impl trace::Span for BoxedSpan {
         self.0.set_attribute(attribute)
     }
 
+    /// Sets attributes of this span.
+    fn set_attributes(&mut self, attributes: Vec<KeyValue>) {
```

Comment:
This should probably take an `impl Iterator<Item = KeyValue>` instead (or would that endanger object safety for the `Span` trait?), so the caller isn't required to allocate a `Vec` for it. I'm not sure about the `set_` prefix, which to me implies overwriting any existing attributes. Maybe `update_attributes()`, and document the behavior wrt overwriting pre-existing attributes in the docstring?

Reply:
We should be able to have this be generic after #637

Reply:
@djc I don't have enough expertise here, can you elaborate how caller could avoid the allocation for `Vec`?

Reply:
The caller might pass in some other iterator, for example something with `.map()` combinator that generates `KeyValue` from some other type. In that case, it would be unfortunate if they have to allocate a `Vec` just to pass their attributes to this method. (In the 2021 edition, which is coming in a few weeks, you could also pass in an array, I think.)

Reply:
@lonewolf3739 any chance you want to implement the above changes? Thanks in advance.

Reply:
I will convert this to draft for now and make the necessary changes when I find time, thanks.